_posts/events/2025-06-24-scipy25-create-python-package.md

```diff
@@ -0,0 +1,235 @@
+
+<figure>
+    <picture>
+    <img src="{{ site.baseurl }}/images/github/codespaces/scipy-open-codespace.gif" alt=".">
+    </picture>
+</figure>
+
+
+## <i class="fa-solid fa-cloud"></i> About GitHub Codespaces: Working in the cloud
+
+
+## What is a codespace
+
+GitHub Codespaces are cloud-based development environments that let you
+code directly in your browser—no local setup needed. They provide fully
+configured, container-based environments connected to your GitHub
+repository.
+
+You can customize Codespaces with `.devcontainer` files to match your
+development setup. The Codespace for this workshop opens in Visual Studio
+Code by default, but you can configure it to use any IDE you prefer.
+
+Learn more in the [GitHub Codespaces docs](https://docs.github.com/en/codespaces/overview).
+
+Codespaces are free for up to 60 hours/month on GitHub’s free plan. Be sure
+to shut yours down after the workshop to conserve your allocated time.
+{: .notice}
+
+
+### How to open a codespace
+
+To open a Codespace,
+
+* Go to your forked SciPy 2025 workshop repository on GitHub,
+* Switch to a branch that you want to work on during the workshop,
+* Click the code drop-down button (where you'd find a link to clone a repo)
+* Click on the Codespaces tab in the drop-down.
+
+Within the Codespaces tab you will see a button: <kbd style="background-color: #2da44e; color: white; padding: 2px 6px; border-radius: 4px;">
+Create codespace on main
+</kbd>. This button launches a Codespace from the currently active branch. It will default to the
+`main` branch if you haven't changed your branch yet.
+
+<figure style="width: 55%; margin: 0 auto;">
+  <picture>
+    <source srcset="{{ site.baseurl }}/images/github/codespaces/create-github-codespace-main.webp"
+            type="image/webp">
+    <img src="{{ site.baseurl }}/images/github/codespaces/create-github-codespace-main.png"
+         alt="Create codespace on main button">
+  </picture>
+</figure>
+
+> <i class="fa-solid fa-circle-info"></i> To follow GitHub best practices, always create a new branch before making
+> changes. Avoid working directly on the `main` branch.
+
+If you switch branches in GitHub, Codespaces will launch from the
+selected, active branch.
+
+<figure>
+  <picture>
+    <img src="{{ site.baseurl }}/images/github/codespaces/scipy-open-codespace.gif"
+         alt="GIF showing Codespace launch process">
+  </picture>
+</figure>
+
+
+> Codespaces are associated with your **personal GitHub account**, even if
+you open one from the pyOpenSci repository. The usage counts toward your
+own GitHub credits.
+{: .notice }
+
+
+### View & manage open codespaces
+
+Once you've opened a Codespace, you can return to it later without needing
+to start from scratch. GitHub will reuse the environment, making it launch
+significantly faster the second time.
+
+If possible, resume the **same Codespace** you started in, rather than
+creating a new one in your fork.
+
+<figure>
```

Comment:
Oh, wait, maybe 3 times. yikes. this entire bottom section needs a rewrite / reorg